File: wirecast/changeshotindex.pl
#!/usr/bin/perl
use strict;
use CGI;

require Win32::OLE;

print "getting wirecast...\n";
my $w = Win32::OLE->GetActiveObject("Wirecast.Application");

if ($w) {
    my $d = $w->DocumentByIndex(1);
    my $l = $d->LayerByName("Normal");
    if($ARGV[0] =~ /^[+-]?\d+$/) {
        my $s = $l->ShotIDByIndex($ARGV[0]);
        if($s) {
            print $s;
            print "attempting to change shot...\n";
            $l->{'ActiveShotID'} = $s;
        }
    }

}else{
    print "couldn't get wirecast";
}

File: wirecast/changeshotindex.js
try {
    //Dependencies
    var win32ole = require('win32ole');
    
    //Should be passed in
    var docID = 1;
    var shotIndex = 2;

    //Get Wirecast
    var wirecast = win32ole.client.Dispatch('Wirecast.Application');
    var document = wirecast.DocumentByIndex(1);
    //changeShotIndex(document.LayerByName('Normal'), shotIndex);

}catch(e){
    console.log('**exception caught ***\n' + e);
}

exports.changeShotIndex = function(layer, index) {
    try{
        var shotID = layer.ShotIDByIndex(index);

        //Using setters and getters is supposed to be obsoleted but calling
        //`layer.ActiveShotID = shotID` causes OLE errors.
        layer.set('ActiveShotID', shotID);
    }catch(e){
        console.log('**exception caught ***\n' + e);
    }
    return true;
}
